dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModel.java

```diff
@@ -337,9 +338,13 @@ private void addEvent(Event e, int globalId, int localId) {
             }
 
             if (data.isRead() || data.isWrite()) {
-                String valueString = String.valueOf(model.evaluate(encodingContext.value((MemoryEvent) e)));
+                Formula valueFormula = encodingContext.value((MemoryCoreEvent)e);
+                assert valueFormula != null;
+                String valueString = String.valueOf(model.evaluate(valueFormula));
                 BigInteger value = switch(valueString) {
-                    case "false" -> BigInteger.ZERO;
+                    // NULL case can happen if the solver optimized away a variable.
+                    // This should only happen if the value is irrelevant, so we will just pick 0.
```

Comment:
Can we log a warning so we can spot when this happen and check that indeed the variable is irrelevant?

Reply:
Sure. I can only log the event though, so it might be hard to connect that back to the source code.